src/arith/scalable_expression.cc

```diff
@@ -50,5 +66,20 @@ std::optional<int> ExtractVscaleFactor(const PrimExpr& lanes) {
   }
 }
 
+bool CanProveVscaleExpressionFromKnownValues(arith::Analyzer* analyzer, const PrimExpr& expr,
+                                             const std::vector<unsigned int>& vscale_values) {
+  bool can_prove_expr = true;
```

Comment:
Out of interest, what happens if the `expr` is not inequality? Should we `ICHECK` for it?